src/ui/pages/equipment/installeqp/InstallEqp.js

```diff
@@ -1,74 +1,134 @@
-import React, {useState} from 'react';
+import React, { useState, useEffect } from 'react';
 import EISPanel from 'eam-components/dist/ui/components/panel';
 import Button from '@material-ui/core/Button';
-import Grid from '@material-ui/core/Grid'
-import WSEquipment from "../../../../tools/WSEquipment";
-import WS from "../../../../tools/WS";
-import EAMBarcodeInput from "eam-components/dist/ui/components/muiinputs/EAMBarcodeInput";
+import Grid from '@material-ui/core/Grid';
+import WSEquipment from '../../../../tools/WSEquipment';
+import WS from '../../../../tools/WS';
+import EAMBarcodeInput from 'eam-components/dist/ui/components/muiinputs/EAMBarcodeInput';
 import EAMAutocomplete from 'eam-components/dist/ui/components/muiinputs/EAMAutocomplete';
 import BlockUi from 'react-block-ui';
+import queryString from "query-string";
 
 export default function NameForm(props) {
-
-    const [parentEq, setParentEq] = useState("");
-    const [childEq, setChildEq] = useState("");
+    const [parentEq, setParentEq] = useState('');
+    const [childEq, setChildEq] = useState('');
     const [blocking, setBlocking] = useState(false);
 
+    useEffect(() => {
+        //Check URL parameters
+        const values = queryString.parse(window.location.search)
+        const parent = values.parent;
+        const child = values.child;
+        //Get all the properties
+        if (parent) {
+            setParentEq(parent);
+        }
+        if (child) {
+            setChildEq(child);
+        }
+    }, [])
+
+    useEffect(() => {
+        if (props.layout.eqpTreeCurrentNode) {
+            setParentEq(props.layout.eqpTreeCurrentNode.id);
+        }
+    }, [props.layout.eqpTreeCurrentNode]);
+
+    useEffect(() => {
+        replaceSearchParams();
+    }, [childEq, parentEq])
+
     const createEquipmentStructure = (newParent, child) => {
         return {
             newParentCode: newParent,
-            childCode: child
+            childCode: child,
         };
+    };
+
+    const replaceSearchParams = () => {
+        let params = '?';
+        if (parentEq) {
+            params += `parent=${parentEq}`;
+        }
+        if (childEq) {
+            if (params !== '?') params += `&`;
+            params += `child=${childEq}`;
+        }
+        if (params === '?') {
+            params = '';
+        }
```

Comment:
why are we not relying on queryString to generate the params string? What if we have some other params? They should not be joined with the parent and child?